### PR TITLE
Feature: Testing framework improvements

### LINKED ISSUE #986: Improvement to the testing library
Creating new Cadl test still require quite a bit of boilerplate. There is a few things we could do to make the life easier of library authors.

## `createTestLibrary`
Each one of the libraries we have(expect the compile) have the same `CadlTestLibrary` definition structure 

```ts
export const OpenAPITestLibrary: CadlTestLibrary = {
  name: "@cadl-lang/openapi",
  packageRoot: resolvePath(fileURLToPath(import.meta.url), "../../../../"),
  files: [
    { realDir: "", pattern: "package.json", virtualPath: "./node_modules/@cadl-lang/openapi" },
    {
      realDir: "dist/src",
      pattern: "*.js",
      virtualPath: "./node_modules/@cadl-lang/openapi/dist/src",
    },
  ],
};
```

could see if we could replace it like that

```ts
export const OpenAPITestLibrary = createTestLibrary("@cadl-lang/openapi");
```

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #986

### CODE REVIEW

File: packages/website/docs/extending-cadl/basics.md [hunk 1]
```diff
@@ -159,3 +159,96 @@ model Employee extends Person {
 ## Next steps
 
 Cadl libraries can contain more than just types. Read the subsequent topics for more details on how to write [decorators](./create-decorators.md), [emitters](./emitters-basics.md) and [linters](./linters.md).
+
+## Testing
+
+Cadl provide a testing framework to help testing library. Examples here are shown using `mocha` but any other JS test framework can be used.
```
Comment (nguerrera) [inline packages/website/docs/extending-cadl/basics.md:165]:
```suggestion
Cadl provides a testing framework to help testing libraries. Examples here are shown using `mocha` but any other JS test framework can be used.
```

**Review by nguerrera — APPROVED** [review]
Nice.

File: docs/extending-cadl/basics.md [hunk 2]
```diff
@@ -159,3 +159,96 @@ model Employee extends Person {
...
+### Define the testing library
+
+First step is to define how your library can be loaded from the test framework. This will let your library to be reused by other library test.
+
+1. Create a new file `./src/testing/index.ts` with the following content
+
+```ts
+export const MyTestLibrary = createTestLibrary({
+  name: "<name-of-npm-pkg>",
+  packageRoot: resolvePath(fileURLToPath(import.meta.url), "../../../../"),
```
Comment (bterlson) [inline docs/extending-cadl/basics.md:176]: Should document what this path is, depending on how your package is laid out you may need a different path here?

File: docs/extending-cadl/basics.md [hunk 3]
```diff
@@ -159,3 +159,96 @@ model Employee extends Person {
...
+    libraries: [RestTestLibrary, MyTestLibrary], // Add other libraries you depend on in your tests
+  });
+}
+export async function createMyTestRunner() {
+  const host = await createOpenAPITestHost();
+  return createTestWrapper(host, { autoUsings: ["My"] });
+}
+```
+
+### Write tests
```
Comment (bterlson) [inline docs/extending-cadl/basics.md:224]: Can perhaps add a small section on the `@test` decorator?

**Review by bterlson — APPROVED** [review]
Very nice!

**Pull request merged**